src/ToricVarieties/AttributesAndMethods.jl

```diff
@@ -1,50 +1,345 @@
-    ambient_dimension = v.polymakeNTV.FAN_AMBIENT_DIM::Int
+    dimension_of_fan = pm_ntv(v).FAN_DIM::Int
+    ambient_dimension = pm_ntv(v).FAN_AMBIENT_DIM::Int
     return ambient_dimension - dimension_of_fan
 end
 export dim_of_torusfactor
 
 
 @doc Markdown.doc"""
-    euler_characteristic( v::AbstractNormalToricVariety )
+    ith_betti_number(v::AbstractNormalToricVariety, i::Int)
+
+Compute the i-th Betti number of the normal toric variety `v`.
+"""
+function ith_betti_number(v::AbstractNormalToricVariety, i::Int)
+    if isodd(i)
+        return 0
+    end
+    k = div(i, 2)
+    f_vector = Vector{Int}(pm_ntv(v).F_VECTOR)
+    pushfirst!(f_vector, 1)
+    betti_number = sum((-1)^(i-k) * binomial(i,k) * f_vector[dim(v) - i + 1] for i=k:dim(v))
+    return betti_number
+    
+end
+export ith_betti_number
+
+
+@doc Markdown.doc"""
+    euler_characteristic(v::AbstractNormalToricVariety)
 
 Computes the Euler characteristic of the normal toric variety `v`.
 """
-function euler_characteristic( v::AbstractNormalToricVariety )
-    f_vector = Vector{Int}(v.polymakeNTV.F_VECTOR)
-    return f_vector[ dim( v ) ]
+function euler_characteristic(v::AbstractNormalToricVariety)
+    f_vector = Vector{Int}(pm_ntv(v).F_VECTOR)
+    return f_vector[dim(v)]
 end
 export euler_characteristic
 
 
+############################
+# Rings and ideals
+############################
+
+
+@doc Markdown.doc"""
+    cox_ring(v::AbstractNormalToricVariety)
+
+Computes the Cox ring of the normal toric variety `v`.
+
+# Examples
+```jdoctest
+julia> p2 = toric_projective_space(2)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> cox_ring(p2)
+(Multivariate Polynomial Ring in x[1], x[2], x[3] over Rational Field graded by 
+  x[1] -> [0 0 1]
+  x[2] -> [0 0 1]
+  x[3] -> [0 0 1], MPolyElem_dec{fmpq, fmpq_mpoly}[x[1], x[2], x[3]])
+```
+"""
+function cox_ring(v::AbstractNormalToricVariety)
+    Qx, x = PolynomialRing(QQ, :x=>1:pm_ntv(v).N_RAYS)
+    return grade(Qx,gens(class_group(v)))[1]
+end
+export cox_ring
+
+
+@doc Markdown.doc"""
+    list_of_variables_of_cox_ring(v::NormalToricVariety)
+
+Lists the names of the indeterminates of the Cox ring of a normal toric variety `v`.
+
+# Examples
+```jdoctest
+julia> p2 = toric_projective_space(2)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> list_of_variables_of_cox_ring(p2)
+3-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
+ x1
+ x2
+ x3
+```
+"""
+function list_of_variables_of_cox_ring(v::AbstractNormalToricVariety)
+    return gens(cox_ring(v))
+end
+export list_of_variables_of_cox_ring
+
+
+@doc Markdown.doc"""
+    stanley_reisner_ideal(v::NormalToricVariety)
+Computes the Stanley-Reisner ideal of a normal toric variety `v`.
+
+# Examples
+```jdoctest
+julia> p2 = toric_projective_space(2)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> length(stanley_reisner_ideal(P2).gens)
```

Comment:
In general it's a bad idea to access struct members directly, this exposes implementation details. Better to do this (assuming it works; perhaps a suitable `gens` method needs to be installed first)
```suggestion
julia> length(gens(stanley_reisner_ideal(P2)))
```
Though actually the Oscar-ish way to do that would be to use `ngens`:
```suggestion
julia> ngens(stanley_reisner_ideal(P2))
```

Reply:
Thank you for the suggestion @fingolfin. `ngens(stanley_reisner_ideal(P2))` does indeed exist. So I'll go with this change.